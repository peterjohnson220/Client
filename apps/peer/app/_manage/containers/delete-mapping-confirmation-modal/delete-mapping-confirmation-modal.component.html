<pf-modal-form
  title="Delete Mapping"
  size=""
  secondaryButtonText="No"
  primaryButtonClass="btn-danger"
  primaryButtonText="Yes"
  primaryButtonTextSubmitting="Deleting..."
  [submitting]="deletingMapping$ | async"
  [isOpen$]="deleteMappingConfirmationModalOpen$"
  (onSubmit)="handleDeleteConfirmed()"
  (onDismiss)="handleDeleteDenied()">

  <ng-container basic-content>
    You are about to delete the association between
    <strong>{{selectedExchangeJobTitle}}</strong> and
    <strong>{{currentCompanyJob?.CompanyJobTitle}}</strong>.
    This cannot be undone.
    <br /><br />
    Would you like to continue?
  </ng-container>

</pf-modal-form>
