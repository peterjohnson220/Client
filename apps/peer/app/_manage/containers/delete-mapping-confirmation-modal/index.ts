export * from './delete-mapping-confirmation-modal.component';
