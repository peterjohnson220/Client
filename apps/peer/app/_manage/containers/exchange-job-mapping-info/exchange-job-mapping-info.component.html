<div class="card mb-2 mapping-info-container">
  <div class="card-header">
    <h6 class="mr-auto mb-0 font-weight-bold">Job Information</h6>
    <ng-container *ngIf="selectedExchangeJobMapping.Mapped">
      <button id="add-mapping-button"
              class="btn btn-sm mr-4"
              [disabled]="selectedExchangeJobMapping.CompanyJobMappings.length > 9"
              [ngClass]="{ 'btn-secondary': !addingMapping, 'btn-warning': addingMapping }"
              (click)="toggleAdding()">
        <span>{{ addingMapping ? 'Cancel' : 'Add Match' }}</span>
      </button>
    </ng-container>
    <a href="javascript:void(0)" (click)="close()" class="close-panel mr-2 mt-1" title="Close">
      <fa-icon icon="times"></fa-icon>
    </a>
  </div>
  <div class="d-flex h-100">
    <div class="w-50 pt-3 pl-3">
      <pf-job-info-container [job]="exchangeJobInfo">
      </pf-job-info-container>
    </div>
    <div class="vertical-separator mt-3 mb-2"></div>
    <div class="w-50 pt-3 pl-3">
      <pf-async-container class="company-job-mapping-container"
                        [opacityLevel]="0.4"
                        [noOpactity]="true"
                        [loading]="(applyingMapping$ | async) || (deletingMapping$ | async)  || (loadingExchangeMappingInfo$ | async)">
        <div class="company-job-mapping d-flex flex-column justify-content-between"
             *ngIf="selectedExchangeJobMapping.Mapped && !addingMapping ">
          <ngb-carousel [activeId]="getActiveCompanyJobId$ | async"
                        [interval]="0"
                        (slide)="onCarouselSlideChange($event)">
            <ng-container *ngFor="let mapping of selectedJobs; trackBy: trackById">
              <ng-template ngbSlide
                           [id]="mapping.CompanyJobId">
                <button class="unmatch float-right mr-3" (click)="handleDeleteClick()">
                  <pf-unmatch-icon></pf-unmatch-icon><span>Unmatch</span>
                </button>
                <button *ngIf="(isAdminOrImpersonating$ | async) || (exchangeStatus$ | async) !== status.Preliminary"
                  class="create-project float-right mr-3"
                  [pfSecuredResource]="permissions.NEW_PROJECT"
                  (click)="handleCreateProjectClick()">
                  <fa-icon [icon]="['far','calculator']" ></fa-icon><span>Create Project</span>
                </button>
                <pf-job-info-container [job]="mapping"
                                       [showJobCode]="true">
                </pf-job-info-container>
              </ng-template>
            </ng-container>
          </ngb-carousel>
          <div class="carousel-controls d-flex flex-row justify-content-between"
               *ngIf="selectedJobs.length > 1">
            <div class="control control-left"
                 (click)="handleControlLeftClick()">
              <fa-icon icon="chevron-circle-left" size="2x"></fa-icon>
            </div>
            <div class="indicators flex-grow-1 space-out-children text-center">
              <span class="control"
                    [ngClass]="{'active' : (getActiveCompanyJobId$ | async) ===  mapping.CompanyJobId}"
                    *ngFor="let mapping of (selectedCompanyJobInfoModels$ | async)"
                    (click)="handleIndicatorClick(mapping.CompanyJobId)">
                <fa-icon icon="circle"></fa-icon>
              </span>
            </div>
            <div class="control control-right"
                 (click)="handleControlRightClick()">
              <fa-icon icon="chevron-circle-right" size="2x"></fa-icon>
            </div>
          </div>
        </div>
        <div class="company-job-mapping-add"
             *ngIf="!selectedExchangeJobMapping.Mapped || addingMapping">
          <h5 class="font-weight-bold mb-4">Search for Company Job</h5>
          <div style="display:flex;">
            <pf-input-debounce [placeholderText]="'Search for a Job Code or Title...'"
                               [(ngModel)]="companyJobQuery"
                               [disabled]="applyingMapping$ | async"
                               (valueChanged)="handleSearchTitleValueChanged($event)"
                               class="search-box">
            </pf-input-debounce>
            <pf-input-debounce [placeholderText]="'Search in the Job Description...'"
                               [(ngModel)]="companyDescriptionQuery"
                               [disabled]="applyingMapping$ | async"
                               (valueChanged)="handleSearchDescValueChanged($event)"
                               class="search-box">
            </pf-input-debounce>
          </div>
          <div class="mapping-results-container"
               [class.remove-scroll]="(companyJobsToMapToLoading$ | async) || (loadingExchangeMappingInfo$ | async)">
            <pf-async-container [loading]="(companyJobsToMapToLoading$ | async) || (loadingExchangeMappingInfo$ | async)"
                                [loadingError]="companyJobsToMapToLoadingError$ | async"
                                [hideReloadButton]="true">
              <ng-container *ngIf="(companyJobsToMapTo$ | async).length">
                <pf-company-job-map-result *ngFor="let companyJobToMapTo of companyJobsToMapTo$ | async"
                                           [companyJobToMapTo]=companyJobToMapTo
                                           [applyingMapping]="applyingMapping$ | async"
                                           [applyingMappingError]="applyingMappingError$ | async"
                                           [selectedMapping]="(selectedMappingCompanyJobId$ | async) == companyJobToMapTo.CompanyJobId"
                                           (applyMapping)="handleApplyMapping($event)"
                                           [highlightFilter]="companyDescriptionQuery">
                </pf-company-job-map-result>
              </ng-container>
              <ng-container *ngIf="(companyJobQuery.length || companyDescriptionQuery.length) && !(companyJobsToMapTo$ | async).length && !(companyJobsToMapToLoading$ | async)">
                <div class="d-flex flex-column mt-5 align-items-center no-search-results">
                  <p class="text-center text-muted pt-4 mb-1">
                    No results for <span class="font-italic">{{ companyJobQuery }} {{ companyDescriptionQuery }}</span>
                  </p>
                  <p class="text-center">
                    <a href="javascript:void(0)" (click)="resetSearchValues()" class="reset font-italic">
                      <fa-icon icon="redo-alt" class="mr-1"></fa-icon>Reset now
                    </a>
                  </p>
                </div>
              </ng-container>
            </pf-async-container>
          </div>
        </div>
      </pf-async-container>
    </div>
  </div>

</div>
<pf-confirm-create-project-modal></pf-confirm-create-project-modal>
<pf-delete-mapping-confirmation-modal [selectedExchangeJobTitle]="selectedExchangeJobMapping.ExchangeJobTitle"
                                      [jobTitle]="selectedCompanyJob?.JobTitle"
                                      [exchangeJobToCompanyJobId]="selectedExchangeJobToCompanyJobId">
</pf-delete-mapping-confirmation-modal>
