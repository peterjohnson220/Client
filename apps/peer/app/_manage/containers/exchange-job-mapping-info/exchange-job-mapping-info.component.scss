@import 'colors';

.card-header{
  height: 38px;
  min-height: 38px;
}

.card-body{
  padding-left: 0;
  padding-right: 0;
  padding-bottom: .25rem;
}

.mapping-info-container{
  position: absolute;
  height: 100%;
  width: 100%;

  .card-header {
    display: flex;
    padding: 0.5rem .5rem 0.5rem 1.25rem;
  }

  .card-body {
    //Firefox requires that flexbox has a min-height to calculate flex-item heights [JP]
    min-height: 0;
  }
}

.mapping-info-container .card-header{
  display: flex;
  padding: 0.5rem .5rem 0.5rem 1.25rem;
}

.mapping-info-container .card-header .btn{
  margin-left: 0.5rem;
}

.remove-scroll{
  overflow: hidden;
}

.company-job-mapping-container {
  flex: 0 0 50%;
  position: relative;

  & > * {
    height: 100%;
  }

  .company-job-mapping-add {
    padding: 0 1rem 0 1rem;
    display: flex;
    flex-direction: column;

    .mapping-results-container {
      margin-top: 0.5rem;
      position: relative;
      height: 100%;
      overflow-y: auto;
    }
  }
}

.close-btn{
  cursor: pointer;
  padding: .25rem 0.75rem;
  margin-left: 0.5rem;
  align-self: center;
}

.close-btn:hover{
  background-color: $selection-hover-background;
}

.delete-mapping-button {
  float: right;
  margin: 0 1rem 0 -4em;
  position: relative;
}

.carousel-controls {
  width: 100%;
  align-items: center;
  padding: .5rem 1rem .5rem 1rem;

  .control {
    color: $inactive-button-color;
    cursor: pointer;
    &:hover {
      color: $body-background-color;
    }
  }

  .control-left, .control-right {
    &:hover {
      color: $body-background-color;
    }
  }

  .indicators > .active {
    color: $body-background-color;
  }
}

:host ::ng-deep ngb-carousel {
  outline: none;
  display: flex !important;
  flex-direction: column;
  height: 100%;

  .carousel-inner {
    display: flex;
  }

  .carousel-item.active {
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  .carousel-control-prev, .carousel-control-next, .carousel-indicators {
    display: none;
  }
}
