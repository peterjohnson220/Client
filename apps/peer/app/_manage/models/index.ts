export * from './exchange-job-request-candidate.model';
