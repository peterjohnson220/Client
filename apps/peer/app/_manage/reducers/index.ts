import { createSelector, createFeatureSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromGridReducer from 'libs/core/reducers/grid.reducer';
import { IFeatureGridState } from 'libs/core/reducers/grid.reducer';

import * as fromExchangeJobMappingGridReducer from './exchange-job-mapping-grid.reducer';
import * as fromExchangeJobMappingInfoReducer from './exchange-job-mapping-info.reducer';
import * as fromPfJobsExchangeRequestReducer from './pf-jobs.reducer';
import * as fromSharedPeerReducer from '../../shared/reducers';
import * as fromExchangeRequestReducer from '../../shared/reducers/exchange-request.reducer';
import * as fromJobFamiliesReducer from './job-families.reducer';

// Feature area state
export interface PeerManagementState {
  exchangeJobMapping: IFeatureGridState<fromExchangeJobMappingGridReducer.State>;
  exchangeJobMappingInfo: fromExchangeJobMappingInfoReducer.State;
  pfJobsExchangeRequest: fromPfJobsExchangeRequestReducer.State;
  jobFamilies: fromJobFamiliesReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  peer_manage: PeerManagementState;
}

// Feature area reducers
export const reducers = {
  exchangeJobMapping: fromExchangeJobMappingGridReducer.reducer,
  exchangeJobMappingInfo: fromExchangeJobMappingInfoReducer.reducer,
  pfJobsExchangeRequest: fromPfJobsExchangeRequestReducer.reducer,
  jobFamilies: fromJobFamiliesReducer.reducer
};

// Select Feature Area
export const selectFeatureAreaState = createFeatureSelector<PeerManagementState>('peer_manage');

// Feature Selectors
export const selectExchangeJobMappingState = createSelector(
  selectFeatureAreaState,
  (state: PeerManagementState) => state.exchangeJobMapping
);

export const selectExchangeJobMappingInfoState = createSelector(
  selectFeatureAreaState,
  (state: PeerManagementState) => state.exchangeJobMappingInfo
);

export const selectPfJobsExchangeRequestState = createSelector(
  selectFeatureAreaState,
  (state: PeerManagementState) => state.pfJobsExchangeRequest
);

export const selectJobFamiliesState = createSelector(
  selectFeatureAreaState,
  (state: PeerManagementState) => state.jobFamilies
);

// Exchange Job Mapping Selectors
export const selectExchangeJobMappingsFeatureState = createSelector(
  selectExchangeJobMappingState,
  (state: IFeatureGridState<fromExchangeJobMappingGridReducer.State>) => state.feature
);

export const selectExchangeJobMappingsGridState = createSelector(
  selectExchangeJobMappingState,
  (state: IFeatureGridState<fromExchangeJobMappingGridReducer.State>) => state.grid
);

export const {
  selectAll: getExchangeJobMappings
} = fromExchangeJobMappingGridReducer.adapter.getSelectors(selectExchangeJobMappingsFeatureState);

export const getExchangeJobMappingsLoading = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getLoading
);

export const getExchangeJobMappingsLoadingError = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getLoadingError
);

export const getExchangeJobMappingsTotal = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getTotal
);

export const getExchangeJobMappingsGridState = createSelector(
  selectExchangeJobMappingsGridState,
  fromGridReducer.getGridState
);

export const getSelectedExchangeJobMapping = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getSelectedMapping
);

export const getExchangeJobMappingPageRowIndexToScrollTo = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getPageRowIndexToScrollTo
);

export const getExchangeJobMappingsGridData = createSelector(
  getExchangeJobMappings,
  getExchangeJobMappingsTotal,
  (data, total) => {
    return { data: data, total: total };
  }
);

export const getLoadExchangeJobMappingGridRequest = createSelector(
  fromSharedPeerReducer.getExchange,
  getExchangeJobMappingsGridState,
  (exchange, gridState) => {
    return { exchangeId: exchange ? exchange.ExchangeId : 0, listState: gridState };
  }
);

// Exchange Job Mapping Info Selectors
export const getCompanyJobsToMapToLoading = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getLoading
);

export const getCompanyJobsToMapToLoadingError = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getLoadingError
);

export const getExchangeJobsInfoApplyingMapping = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getApplyingMapping
);

export const getExchangeJobsInfoApplyingMappingError = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getApplyingMappingError
);

export const getExchangeJobsInfoSelectedMappingCompanyJobId = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getSelectedMappingCompanyJobId
);

export const getExchangeJobsInfoEditingMapping = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getEditingMapping
);

export const {
  selectAll: getCompanyJobsToMapTo
} = fromExchangeJobMappingInfoReducer.adapter.getSelectors(selectExchangeJobMappingInfoState);

// Exchange Request - pfJobs
export const {
  selectAll: getPfJobsExchangeRequestCandidates
} = fromPfJobsExchangeRequestReducer.adapter.getSelectors(selectPfJobsExchangeRequestState);
export const getPfJobsExchangeRequestCandidatesLoading = createSelector(
  selectPfJobsExchangeRequestState,
  fromExchangeRequestReducer.getLoading
);
export const getPfJobsExchangeRequestCandidatesLoadingError = createSelector(
  selectPfJobsExchangeRequestState,
  fromExchangeRequestReducer.getLoadingError
);
export const getPfJobsExchangeRequestModalOpen = createSelector(
  selectPfJobsExchangeRequestState,
  fromExchangeRequestReducer.getModalOpen
);
export const getPfJobsExchangeRequestRequesting = createSelector(
  selectPfJobsExchangeRequestState,
  fromExchangeRequestReducer.getRequesting
);
export const getPfJobsExchangeRequestContext = createSelector(
  selectPfJobsExchangeRequestState,
  fromSharedPeerReducer.getExchange,
  (exchangeRequestState, exchange) => {
    return {
      Query: exchangeRequestState.searchTerm,
      ExchangeId: exchange ? exchange.ExchangeId : 0,
      JobDescriptionQuery: exchangeRequestState.filterOptions ? exchangeRequestState.filterOptions.JobDescriptionQuery : ''
    };
  }
);

// Job Families
export const {
  selectAll: getJobFamilies
} = fromJobFamiliesReducer.adapter.getSelectors(selectJobFamiliesState);
export const getJobFamiliesLoading = createSelector(
  selectJobFamiliesState,
  fromJobFamiliesReducer.getLoading
);
export const getJobFamiliesLoadingError = createSelector(
  selectJobFamiliesState,
  fromJobFamiliesReducer.getLoadingError
);
