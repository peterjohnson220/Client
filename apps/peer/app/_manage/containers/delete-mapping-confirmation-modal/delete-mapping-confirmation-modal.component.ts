import { Component, Input } from '@angular/core';

import { Store } from '@ngrx/store';
import { Observable } from 'rxjs';

import { ExchangeJobMapping } from 'libs/models';

import * as fromExchangeJobMappingInfoActions from '../../actions/exchange-job-mapping-info.actions';
import * as fromPeerManagementReducer from '../../reducers';

@Component({
  selector: 'pf-delete-mapping-confirmation-modal',
  templateUrl: './delete-mapping-confirmation-modal.component.html',
  styleUrls: [ './delete-mapping-confirmation-modal.component.scss' ]
})
export class DeleteMappingConfirmationModalComponent {
  @Input() selectedExchangeJobMapping: ExchangeJobMapping;

  deleteMappingConfirmationModalOpen$: Observable<boolean>;
  deletingMapping$: Observable<boolean>;
  deletingMappingError$: Observable<boolean>;

  constructor(private store: Store<fromPeerManagementReducer.State>) {
    this.deleteMappingConfirmationModalOpen$ = this.store.select(fromPeerManagementReducer.getExchangeJobsInfoDeleteConfirmationModalOpen);
    this.deletingMapping$ = this.store.select(fromPeerManagementReducer.getExchangeJobsInfoDeletingMapping);
    this.deletingMappingError$ = this.store.select(fromPeerManagementReducer.getExchangeJobsInfoDeletingMappingError);
  }

  // Modal events
  handleDeleteConfirmed() {
    this.store.dispatch(new fromExchangeJobMappingInfoActions.DeleteMapping({
        exchangeJobToCompanyJobId: this.selectedExchangeJobMapping.ExchangeJobToCompanyJobId
      }
    ));
  }

  handleDeleteDenied() {
    this.store.dispatch(new fromExchangeJobMappingInfoActions.CloseDeleteConfirmationModal());
  }
}
