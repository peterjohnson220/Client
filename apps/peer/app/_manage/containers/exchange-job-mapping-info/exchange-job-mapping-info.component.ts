import { Component, Input, Output, EventEmitter, OnInit, OnDestroy, ViewChild } from '@angular/core';

import { Store, select } from '@ngrx/store';
import { Observable, Subscription } from 'rxjs';
import { NgbSlideEvent } from '@ng-bootstrap/ng-bootstrap/carousel/carousel';
import { NgbCarousel } from '@ng-bootstrap/ng-bootstrap';
import * as lodash from 'lodash';

import { CompanyJobToMapTo, ExchangeJobMapping, Job, CompanyJobMapping } from 'libs/models';

import * as fromExchangeJobMappingInfoActions from '../../actions/exchange-job-mapping-info.actions';
import * as fromPeerManagementReducer from '../../reducers';

@Component({
  selector: 'pf-exchange-job-mapping-info',
  templateUrl: './exchange-job-mapping-info.component.html',
  styleUrls: ['./exchange-job-mapping-info.component.scss']
})
export class ExchangeJobMappingInfoComponent implements OnInit, OnDestroy {
  @Input() exchangeId: number;
  @Output() closeClicked = new EventEmitter();
  @Output() editMappingClicked = new EventEmitter();
  @ViewChild(NgbCarousel) carousel: NgbCarousel;

  // Observables
  selectedExchangeJobMapping$: Observable<ExchangeJobMapping>;
  companyJobsToMapTo$: Observable<CompanyJobToMapTo[]>;
  companyJobsToMapToLoading$: Observable<boolean>;
  companyJobsToMapToLoadingError$: Observable<boolean>;
  applyingMapping$: Observable<boolean>;
  applyingMappingError$: Observable<boolean>;
  selectedMappingCompanyJobId$: Observable<number>;
  activeExchangeJobToCompanyJobId$: Observable<number>;
  addingMapping$: Observable<boolean>;
  deletingMapping$: Observable<boolean>;
  deletingMappingError$: Observable<boolean>;
  loadingExchangeJobMappings$: Observable<boolean>;

  // Subscriptions
  selectedExchangeJobMappingSubscription: Subscription;
  addingMappingSubscription: Subscription;
  activeExchangeJobToCompanyJobIdSubscription: Subscription;

  selectedExchangeJobMapping: ExchangeJobMapping;
  selectedCompanyJobMapping: CompanyJobMapping;
  selectedCompanyJobInfoModels: Job[];
  exchangeJobInfo: Job;
  companyJobQuery: string;
  companyDescriptionQuery: string;
  debouncedQueryValue: string;
  addingMapping: boolean;

  constructor(private store: Store<fromPeerManagementReducer.State>) {
    this.loadingExchangeJobMappings$ = this.store.pipe(select(fromPeerManagementReducer.getExchangeJobMappingsLoading));
    this.selectedExchangeJobMapping$ = this.store.pipe(select(fromPeerManagementReducer.getSelectedExchangeJobMapping));
    this.companyJobsToMapTo$ = this.store.pipe(select(fromPeerManagementReducer.getCompanyJobsToMapTo));
    this.companyJobsToMapToLoading$ = this.store.pipe(select(fromPeerManagementReducer.getCompanyJobsToMapToLoading));
    this.companyJobsToMapToLoadingError$ = this.store.pipe(select(fromPeerManagementReducer.getCompanyJobsToMapToLoadingError));
    this.applyingMapping$ = this.store.pipe(select(fromPeerManagementReducer.getExchangeJobsInfoApplyingMapping));
    this.applyingMappingError$ = this.store.pipe(select(fromPeerManagementReducer.getExchangeJobsInfoApplyingMappingError));
    this.selectedMappingCompanyJobId$ = this.store.pipe(select(fromPeerManagementReducer.getExchangeJobsInfoSelectedMappingCompanyJobId));
    this.addingMapping$ = this.store.pipe(select(fromPeerManagementReducer.getExchangeJobsInfoAddingMapping));
    this.deletingMapping$ = this.store.pipe(select(fromPeerManagementReducer.getExchangeJobsInfoDeletingMapping));
    this.deletingMappingError$ = this.store.pipe(select(fromPeerManagementReducer.getExchangeJobsInfoDeletingMappingError));
    this.activeExchangeJobToCompanyJobId$ = this.store.pipe(select(fromPeerManagementReducer.getExchangeJobsInfoActiveMapping));
  }

  searchChanged() {
    this.debouncedQueryValue = this.companyJobQuery;

    if (this.companyJobQuery.length <= 0 && this.companyDescriptionQuery.length <= 0) {
      this.debouncedQueryValue = this.selectedExchangeJobMapping.ExchangeJobTitle;
    }

    this.store.dispatch(new fromExchangeJobMappingInfoActions.LoadCompanyJobsToMapToByQuery({
      exchangeId: this.exchangeId,
      jobTitleAndCodeQuery: this.debouncedQueryValue,
      jobDescriptionQuery: this.companyDescriptionQuery
    }));
  }

  handleSearchDescValueChanged(event: string) {
    this.companyDescriptionQuery = event;
    this.searchChanged();
  }

  handleSearchTitleValueChanged(event: string) {
    this.companyJobQuery = event;
    this.searchChanged();
  }

  handleApplyMapping(companyJobId: number) {
    this.store.dispatch(new fromExchangeJobMappingInfoActions.ApplyMapping({
      ExchangeId: this.exchangeId,
      ExchangeJobId: this.selectedExchangeJobMapping.ExchangeJobId,
      CompanyJobId: companyJobId
    }));
  }

  handleDeleteClick() {
    this.store.dispatch(new fromExchangeJobMappingInfoActions.OpenDeleteConfirmationModal());
  }

  resetValues() {
    this.companyJobQuery = '';
    this.companyDescriptionQuery = '';
    this.debouncedQueryValue = '';
  }

  toggleAdding() {
    if (this.addingMapping) {
      this.store.dispatch(new fromExchangeJobMappingInfoActions.CancelAddMapping());
    } else {
      this.resetValues();
      this.dispatchLoadCompanyJobsToMapToByQuery();
      this.store.dispatch(new fromExchangeJobMappingInfoActions.AddMapping());
    }
  }

  close() {
    this.store.dispatch(new fromExchangeJobMappingInfoActions.CancelAddMapping());
    this.closeClicked.emit();
  }

  trackById(index: number, mappedCompanyJob: CompanyJobMapping): number {
    return mappedCompanyJob.ExchangeJobToCompanyJobId;
  }

  // Carousel handlers
  handleControlRightClick() {
    this.carousel.next();
  }

  handleControlLeftClick() {
    this.carousel.prev();
  }

  handleIndicatorClick(index: string) {
    this.carousel.select(index);
  }

  onCarouselSlideChange(slideEvent: NgbSlideEvent): void {
    this.store.dispatch(new fromExchangeJobMappingInfoActions.SetActiveMapping(+slideEvent.current));
  }
  // Lifecycle
  ngOnInit() {
    this.selectedExchangeJobMappingSubscription = this.selectedExchangeJobMapping$.subscribe(sm => {
      if (sm) {
        this.selectedExchangeJobMapping = sm;
        this.resetValues();

        this.store.dispatch(new fromExchangeJobMappingInfoActions.CancelAddMapping());
        this.buildJobModels(sm);

        if (!sm.Mapped) {
          this.dispatchLoadCompanyJobsToMapToByQuery();
        }
      }
    });

    this.activeExchangeJobToCompanyJobIdSubscription = this.activeExchangeJobToCompanyJobId$.subscribe(id => {
      if (id) {
        this.selectedCompanyJobMapping = this.selectedExchangeJobMapping.CompanyJobMappings.find(
          cjm => cjm.ExchangeJobToCompanyJobId === id
        );
      }
    });

    this.addingMappingSubscription = this.addingMapping$.subscribe(em => this.addingMapping = em);
  }

  buildNoResultsString(): string {

    let s = 'No results for ';

    const companyJob = lodash.escape(this.companyJobQuery);
    const jobDescription = lodash.escape(this.companyDescriptionQuery);

    if (!companyJob && !jobDescription) {
      s += '<u>' + this.selectedExchangeJobMapping.ExchangeJobTitle + '</u>';
    } else if (companyJob) {
      s += '<u>' + companyJob + '</u>';
      if (jobDescription) {
        s += ' <i>and</i> ';
      }
    }

    if (jobDescription) {
      s += '<u>' + jobDescription + '</u>';
    }
    return s;

  }

  ngOnDestroy() {
    this.selectedExchangeJobMappingSubscription.unsubscribe();
    this.addingMappingSubscription.unsubscribe();
    this.activeExchangeJobToCompanyJobIdSubscription.unsubscribe();
  }

  private dispatchLoadCompanyJobsToMapToByQuery(): void {
    this.store.dispatch(new fromExchangeJobMappingInfoActions.LoadCompanyJobsToMapToByQuery({
      exchangeId: this.exchangeId,
      jobTitleAndCodeQuery: this.selectedExchangeJobMapping.ExchangeJobTitle,
      jobDescriptionQuery: ''
    }));
  }

  private buildJobModels(ejm: ExchangeJobMapping): void {
    this.exchangeJobInfo = {
      JobType: 'Exchange',
      JobTitle: ejm.ExchangeJobTitle,
      JobCode: ejm.ExchangeJobCode,
      JobFamily: ejm.ExchangeJobFamily,
      JobLevel: ejm.ExchangeJobLevel,
      JobDescription: ejm.ExchangeJobDescription
    };
    this.selectedCompanyJobInfoModels = ejm.CompanyJobMappings.map(this.buildCompanyJobInfoModel);
  }

  private buildCompanyJobInfoModel(cji: CompanyJobMapping): Job {
    return {
      JobId: cji.ExchangeJobToCompanyJobId,
      JobType: 'Company',
      JobTitle: cji.CompanyJobTitle,
      JobCode: cji.CompanyJobCode,
      JobFamily: cji.CompanyJobFamily,
      JobLevel: cji.CompanyJobLevel,
      JobDescription: cji.CompanyJobDescription
    };
  }
}
