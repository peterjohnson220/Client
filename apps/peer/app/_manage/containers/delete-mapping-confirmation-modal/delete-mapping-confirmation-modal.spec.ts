import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';

import { Store, combineReducers, StoreModule } from '@ngrx/store';

import * as fromRootState from 'libs/state/state';
import { generateMockExchangeJobMapping } from 'libs/models/peer';

import * as fromPeerManagementReducer from '../../reducers';
import * as fromExchangeJobMappingInfoActions from '../../actions/exchange-job-mapping-info.actions';
import { DeleteMappingConfirmationModalComponent } from './delete-mapping-confirmation-modal.component';


describe('Peer - Manage - Delete Confirmation Modal', () => {
  let fixture: ComponentFixture<DeleteMappingConfirmationModalComponent>;
  let instance: DeleteMappingConfirmationModalComponent;

  let store: Store<fromPeerManagementReducer.State>;

  const mockExchangeJobMapping = generateMockExchangeJobMapping();

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          peer_manage: combineReducers(fromPeerManagementReducer.reducers)
        }),
      ],
      declarations: [
        DeleteMappingConfirmationModalComponent
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    store = TestBed.inject(Store);

    fixture = TestBed.createComponent(DeleteMappingConfirmationModalComponent);
    instance = fixture.componentInstance;
    instance.selectedExchangeJobTitle = mockExchangeJobMapping.ExchangeJobTitle;

    jest.spyOn(store, 'dispatch');

    // Trigger ngOnInit
    fixture.detectChanges();
  });

  it(`should match snapshot on init`, () => {
    expect(fixture).toMatchSnapshot();
  });

  it(`should dispatch DeleteMapping action when handleDeleteConfirmed is triggered`, () => {
    const mockPayload = {exchangeJobToCompanyJobId: mockExchangeJobMapping.CompanyJobMappings[0].ExchangeJobToCompanyJobId};
    const expectedAction = new fromExchangeJobMappingInfoActions.DeleteMapping(mockPayload);

    instance.exchangeJobToCompanyJobId = mockExchangeJobMapping.CompanyJobMappings[0].ExchangeJobToCompanyJobId;
    instance.handleDeleteConfirmed();

    expect(store.dispatch).toBeCalledWith(expectedAction);
  });

  it(`should dispatch CloseDeleteConfirmationModal action when handleDeleteDenied is triggered`, () => {
    const expectedAction = new fromExchangeJobMappingInfoActions.CloseDeleteConfirmationModal();

    instance.handleDeleteDenied();

    expect(store.dispatch).toBeCalledWith(expectedAction);
  });
});
