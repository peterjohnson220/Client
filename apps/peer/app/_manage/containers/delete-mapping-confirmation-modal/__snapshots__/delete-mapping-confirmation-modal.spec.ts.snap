// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Peer - Manage - Delete Confirmation Modal should match snapshot on init 1`] = `
<pf-delete-mapping-confirmation-modal
  __ngContext__={[Function LRootView]}
  deleteMappingConfirmationModalOpen$={[Function Store]}
  deletingMapping$={[Function Store]}
  deletingMappingError$={[Function Store]}
  selectedExchangeJobTitle={[Function String]}
  store={[Function Store]}
>
  <pf-modal-form
    primarybuttonclass="btn-danger"
    primarybuttontext="Yes"
    primarybuttontextsubmitting="Deleting..."
    secondarybuttontext="No"
    size=""
    title="Delete Association"
  >
     You are about to delete the association between 
    <strong>
      Exchange Test Job
    </strong>
     and 
    <strong>
    </strong>
    . This cannot be undone. 
    <br />
    <br />
     Would you like to continue? 
  </pf-modal-form>
</pf-delete-mapping-confirmation-modal>
`;
